### PR TITLE
web-eid-app: init at 2.2.0

### PR DESCRIPTION
**Pull request by flokli**
This adds `web-eid-app`, which is the native companion to the "Web eID" browser extension.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/security/web-eid-app/default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
+
+  buildInputs = [
+    gtest
```
Comment (delroth) [inline pkgs/tools/security/web-eid-app/default.nix:30]: Shouldn't that be a checkInput?

File: pkgs/tools/security/web-eid-app/default.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    pcsclite
+    qttranslations
+  ];
+
+  meta = with lib; {
+    longDescription = ''
```
Comment (delroth) [inline pkgs/tools/security/web-eid-app/default.nix:36]: Could you add a (short) description as well? search.nixos.org doesn't seem to use the longDescription if a description is absent. e.g. libao/libaosd on https://search.nixos.org/packages?channel=22.11&from=0&size=50&sort=relevance&type=packages&query=libao

**Review by delroth — APPROVED** [review]
Two small nits, otherwise LGTM.

File: pkgs/tools/security/web-eid-app/default.nix (hunk 1 above)
Comment (delroth) [reply pkgs/tools/security/web-eid-app/default.nix:30]: Actually, looking further I don't even see a reference to gtest in the code anymore, only in upstream's CI configuration. Could you double check if this is even needed?

Comment (flokli) [reply pkgs/tools/security/web-eid-app/default.nix:30]: This is a dependency by `lib/libelectronic-id/lib/libpcsc-cpp/CMakeLists.txt:68`. If I remove it, I get the following error:

```
CMake Error at /nix/store/9d67jcm1wj09wqgffh8v5mkgil9psnrw-cmake-3.25.3/share/cmake-3.25/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find GTest (missing: GTEST_LIBRARY GTEST_INCLUDE_DIR
  GTEST_MAIN_LIBRARY)
Call Stack (most recent call first):
  /nix/store/9d67jcm1wj09wqgffh8v5mkgil9psnrw-cmake-3.25.3/share/cmake-3.25/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  /nix/store/9d67jcm1wj09wqgffh8v5mkgil9psnrw-cmake-3.25.3/share/cmake-3.25/Modules/FindGTest.cmake:270 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
  lib/libelectronic-id/lib/libpcsc-cpp/CMakeLists.txt:68 (find_package)

```

File: pkgs/tools/security/web-eid-app/default.nix (hunk 2 above)
Comment (flokli) [reply pkgs/tools/security/web-eid-app/default.nix:37]: Updated.

File: pkgs/tools/security/web-eid-app/default.nix (hunk 1 above)
Comment (delroth) [reply pkgs/tools/security/web-eid-app/default.nix:30]: Ah, thanks for the pointer -- GitHub code search doesn't recurse through submodules apparently...

Comment (flokli) [reply pkgs/tools/security/web-eid-app/default.nix]: I added a small comment, in case someone stumbled over this again.

**Comment by flokli** [comment]
Thanks!

**Pull request merged**